Capstone_data_clean.ipynb

```diff
@@ -296,390 +422,727 @@
-    "    data=prcp.find('td')\n",
-    "    nxt=data.find('p')\n",
-    "    point=nxt.find(class_='value')\n",
-    "    precipitation[i]=point.get_text()\n",
-    "    \n",
-    "precipitation=pd.Series(precipitation)\n",
-    "precipitation"
+    "pracdf.head()"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": []
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": []
+   "execution_count": 22,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2019-06-26T20:26:07.961624Z",
+     "start_time": "2019-06-26T20:26:07.946520Z"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "col 0: schedule_date\n",
+      "col 1: schedule_season\n",
+      "col 2: schedule_week\n",
+      "col 3: team_home\n",
+      "col 4: score_home\n",
+      "col 5: score_away\n",
+      "col 6: team_away\n",
+      "col 7: team_favorite_id\n",
+      "col 8: spread_favorite\n",
+      "col 9: results\n",
+      "col 10: results-spread\n",
+      "col 11: favorite_cover\n",
+      "col 12: over_under_line\n",
+      "col 13: total\n",
+      "col 14: over_diff\n",
+      "col 15: over_cover\n",
+      "col 16: weather_temperature\n",
+      "col 17: weather_wind_mph\n",
+      "col 18: stadium\n",
+      "col 19: coordinates\n",
+      "col 20: zipcode\n"
+     ]
+    }
+   ],
+   "source": [
+    "for i, col in enumerate(pracdf.columns):\n",
+    "    print('col {}: {}'.format(i, col))"
+   ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2019-06-26T20:46:18.692897Z",
+     "start_time": "2019-06-26T20:45:31.520901Z"
+    }
+   },
    "outputs": [],
-   "source": []
+   "source": [
+    "#create dic to store values\n",
+    "precipitation = {}\n",
+    "\n",
+    "# loop through index and populate dict with precipitation data\n",
+    "# TODO: look into how to efficiently loop over a dataframe\n",
+    "for i in pracdf.index:\n",
+    "    # temp debug continue\n",
+    "    #if pracdf.loc[i]['team_home'] != 'DET':\n",
+    "    #    continue\n",
+    "    print('Working on: {} -- {} v {}'.format(pracdf.loc[i]['schedule_date'], pracdf.loc[i]['team_home'], pracdf.loc[i]['team_away']))\n",
+    "    page = r.get('https://www.almanac.com/weather/history/zipcode/' +\n",
+    "                 pracdf.loc[i]['zipcode'] + '/' +\n",
+    "                 pracdf.loc[i]['schedule_date'])\n",
+    "    soup = BeautifulSoup(page.content, 'html.parser')\n",
+    "    prcp = soup.find('tr', class_='weatherhistory_results_datavalue prcp')\n",
+    "    data = prcp.find('td')\n",
+    "    nxt = data.find('p')\n",
+    "    #print('---->>>>>', )\n",
+    "    #print('---->>>>>', nxt.find(class_='nullvalue'))\n",
+    "    if nxt.get_text() == 'No data.':\n",
```

Comment:
Recall, this is the if-else block that catches the NoneType err and allows the processing to proceed.